app/mixins/pagination.js

```diff
@@ -10,6 +10,32 @@ export default Ember.Mixin.create({
         return pages;
     }.property('availablePages'),
 
+    // Gives page numbers to the surrounding 9 pages.
+    surroundingPages: function() {
+        var pages = [];
+        var currentPage = this.get('currentPage');
+        var availablePages = this.get('availablePages');
+        var lowerBound = 0;
+        var upperBound = 0;
+
+        // Always show 9 pages even if we're at the beginning or
+        // end of the list.
+        if (availablePages - currentPage < 5) {
+            lowerBound = Math.max(0, availablePages - 9);
+            upperBound = availablePages;
+        } else if (currentPage < 6) {
+            lowerBound = 0;
+            upperBound = Math.min(availablePages, 9);
+        } else {
+            lowerBound = currentPage - 5;
+            upperBound = currentPage + 4;
+        }
+        for (var i = lowerBound; i < upperBound; i++) {
+            pages.push(i + 1);
+        }
+        return pages;
+    }.property('currentPage', 'availablePages'),
```

Comment:
Would it be possible to refactor this such that the number of pages is a constant listed in one location? In theory all the other numbers would be based off that one constant.